crates/ruff_linter/src/linter.rs

```diff
@@ -221,36 +232,56 @@ pub fn check_path(
                 continue;
             }
             let diagnostic = match test_rule {
-                Rule::StableTestRule => test_rules::StableTestRule::diagnostic(locator, indexer),
-                Rule::StableTestRuleSafeFix => {
-                    test_rules::StableTestRuleSafeFix::diagnostic(locator, indexer)
+                Rule::StableTestRule => {
+                    test_rules::StableTestRule::diagnostic(locator, program.comment_ranges())
                 }
-                Rule::StableTestRuleUnsafeFix => {
-                    test_rules::StableTestRuleUnsafeFix::diagnostic(locator, indexer)
+                Rule::StableTestRuleSafeFix => {
```

Comment:
Is the change from `UnsafeFix` to `SafeFix` intentional?

Reply:
No, it's the way diff is being rendered. I've updated the code to assign it to a variable which reduces the diff.